contracts/oracle/FastPriceFeed.sol

```diff
@@ -236,80 +303,152 @@ contract FastPriceFeed is ISecondaryPriceFeed, IFastPriceFeed, Governable {
+    // - in case watchers flag an issue
+    // - in case the cumulativeFastDelta exceeds the cumulativeRefDelta by the maxCumulativeDeltaDiff
     function getPrice(address _token, uint256 _refPrice, bool _maximise) external override view returns (uint256) {
-        if (block.timestamp > lastUpdatedAt.add(priceDuration)) { return _refPrice; }
+        if (block.timestamp > lastUpdatedAt.add(maxPriceUpdateDelay)) {
+            if (_maximise) {
+                return _refPrice.mul(BASIS_POINTS_DIVISOR.add(spreadBasisPointsIfChainError)).div(BASIS_POINTS_DIVISOR);
+            }
+
+            return _refPrice.mul(BASIS_POINTS_DIVISOR.sub(spreadBasisPointsIfChainError)).div(BASIS_POINTS_DIVISOR);
+        }
+
+        if (block.timestamp > lastUpdatedAt.add(priceDuration)) {
+            if (_maximise) {
+                return _refPrice.mul(BASIS_POINTS_DIVISOR.add(spreadBasisPointsIfInactive)).div(BASIS_POINTS_DIVISOR);
+            }
+
+            return _refPrice.mul(BASIS_POINTS_DIVISOR.sub(spreadBasisPointsIfInactive)).div(BASIS_POINTS_DIVISOR);
+        }
 
         uint256 fastPrice = prices[_token];
         if (fastPrice == 0) { return _refPrice; }
 
-        uint256 maxPrice = _refPrice.mul(BASIS_POINTS_DIVISOR.add(maxDeviationBasisPoints)).div(BASIS_POINTS_DIVISOR);
-        uint256 minPrice = _refPrice.mul(BASIS_POINTS_DIVISOR.sub(maxDeviationBasisPoints)).div(BASIS_POINTS_DIVISOR);
-
-        if (favorFastPrice()) {
-            if (fastPrice >= minPrice && fastPrice <= maxPrice) {
-                if (_maximise) {
-                    if (_refPrice > fastPrice) {
-                        uint256 volPrice = fastPrice.mul(BASIS_POINTS_DIVISOR.add(volBasisPoints)).div(BASIS_POINTS_DIVISOR);
-                        // the volPrice should not be more than _refPrice
-                        return volPrice > _refPrice ? _refPrice : volPrice;
-                    }
-                    return fastPrice;
-                }
+        uint256 diffBasisPoints = _refPrice > fastPrice ? _refPrice.sub(fastPrice) : fastPrice.sub(_refPrice);
+        diffBasisPoints = diffBasisPoints.mul(BASIS_POINTS_DIVISOR).div(_refPrice);
 
-                if (_refPrice < fastPrice) {
-                    uint256 volPrice = fastPrice.mul(BASIS_POINTS_DIVISOR.sub(volBasisPoints)).div(BASIS_POINTS_DIVISOR);
-                    // the volPrice should not be less than _refPrice
-                    return volPrice < _refPrice ? _refPrice : volPrice;
-                }
+        // create a spread between the _refPrice and the fastPrice if the maxDeviationBasisPoints is exceeded
+        // or if watchers have flagged an issue with the fast price
+        bool hasSpread = !favorFastPrice(_token) || diffBasisPoints > maxDeviationBasisPoints;
 
-                return fastPrice;
+        if (hasSpread) {
+            // return the higher of the two prices
+            if (_maximise) {
+                return _refPrice > fastPrice ? _refPrice : fastPrice;
```

Comment:
oh no more limit for spread? so if price deviation is 10%, then spread is 10%, right?